packages/protocol/src/parse.ts

```diff
@@ -0,0 +1,118 @@
+import type { MessageModel, ResourceModel, RfqMetadata, RfqData, QuoteData, QuoteMetadata, OrderData, OrderMetadata, OrderStatusMetadata, OrderStatusData, CloseMetadata, CloseData, OfferingMetadata, OfferingData } from './types.js'
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { Resource } from './resource.js'
+import type { Message } from './message.js'
+
+import { Rfq, Quote, Order, OrderStatus, Close } from './message-kinds/index.js'
+import { Offering } from './resource-kinds/index.js'
+
+/**
+ * @beta
+ *
+ * Parses the json message into a message instance.
+ * Performs format validation and an integrity check on the signature
+ * @param message - the message to parse. can either be an object or a string
+ * @returns {@link Message}
+ */
+export async function parseMessage(rawMessage: MessageModel | string): Promise<Message> {
+  const jsonMessage = rawToMessageModel(rawMessage)
+
+  let message: Message
+  switch(jsonMessage.metadata.kind) {
+    case 'rfq':
+      message = new Rfq(
```

Comment:
```suggestion
      message = await Rfq.parse(rawMessage)
```

just wondering if we can use `parse()` method you already wrote for each of the concrete message impls?

Reply:
I'd prefer not to reuse `Rfq.parse` etc. `Rfq.parse()` does some of the same things that are already done in `parseMessage`, so we would be repeating operations (`.verify()`, and `rawToMessageModel`). While we could remove those operations from `parseMessage`, I think it would be less clear what `parseMessage` is doing by putting those things in a method. Also, i prefer to think of `Rfq.parse()` and `parseMessage` as part of the public API for this library, and that we use the constructors and instance methods when appropriate internally.